storage/tests/unit/test_blob.py

```diff
@@ -582,6 +614,27 @@ def test_exists_miss(self):
             },
         )
 
+    def test_exists_miss_within_batch(self):
+        from google.cloud.storage.client import Client
+        from google.cloud.exceptions import NotFound
+
+        client = Client(credentials=_make_credentials())
+        client._base_connection = _make_connection()
+
+        bucket = _Bucket(client)
+        blob1 = self._make_one("nonesuch1", bucket=bucket)
+        blob2 = self._make_one("nonesuch2", bucket=bucket)
+
+        try:
+            with client.batch():
+                bool1 = blob1.exists()
+                bool2 = blob2.exists()
+        except NotFound:
+            pass
+
+        self.assertFalse(bool1)
+        self.assertTrue(bool2)
```

Comment:
Checking if both `exists()` calls working as they suppose to: the first one must be `False` (NotFound)